code/modules/mob/mob_movement.dm

```diff
@@ -328,6 +328,8 @@
 					return AM
 				if(pulling == AM)
 					continue
+				if(get_turf(AM) == get_step(get_turf(src), src.dir)) //no moving fowards to push off and obj and also push it fowards
```

Comment:
The comment does not seem to make a valid english sentence (or maybe i'm just that tired)... Can you try rewording it to be easier to read?